proxy.ts

```diff
@@ -40,4 +23,43 @@ export function getProxyUrl(reqUrl: url.Url): url.Url {
     }
 
     return proxyUrl;
+}
+
+
+export function checkBypass(reqUrl: url.Url): boolean {
+    if (!reqUrl.hostname) {
+        return false
+    }
+
+    let noProxy = process.env["no_proxy"] || process.env["NO_PROXY"] || '';
+    if (!noProxy) {
+        return false
+    }
+
+    // Determine the request port
+    let reqPort: number
+    if (reqUrl.port) {
+        reqPort = Number(reqUrl.port)
+    }
+    else if (reqUrl.protocol === 'http:') {
+        reqPort = 80
+    }
+    else if (reqUrl.protocol === 'https:') {
+        reqPort = 443
+    }
+
+    // Format the request hostname and hostname with port
```

Comment:
The trick here is generate two host values to check against the no_proxy list.

When reqUrl is like `https://foo.com` this will create two strings to check against the bypass list: `FOO.COM` and `FOO.COM:443`